R/canadacovidmetrics.R

```diff
@@ -90,7 +90,30 @@ total_cumulative_cases <- function(loc='prov', date=NULL, after='2020-01-01', be
 #' total_cumulative_deaths(loc = "ON", before = "2021-12-31")
 #' total_cumulative_deaths(loc = "prov", date = "2021-09-01")
 
-total_cumulative_deaths <- function(loc='prov', date=NULL, after='2020-01-01', before=Sys.Date()){}
+total_cumulative_deaths <- function(loc='prov', date=NULL, after='2020-01-01', before=Sys.Date(), datetime_type=FALSE){
+  
+  loc_format_check(loc)
+  
+  if (!is.null(date)) {
+    date_format_check(date)
+    url <- sprintf("https://api.opencovid.ca/timeseries?stat=mortality&loc=%s&date=%s", loc, date)
+  } else {
+    date_format_check(after)
+    date_format_check(before)
+    url <- sprintf("https://api.opencovid.ca/timeseries?stat=mortality&loc=%s&after=%s&before=%s", loc, after, before)
+  }
+  
+  raw_json <- GET(url)
+  json_string <- content(raw_json, as="text")
+  df <- fromJSON(json_string)$mortality
+
+  if(datetime_type){
+    df$date_vaccine_completed = as.Date(df$date_vaccine_completed, format = "%d-%m-%y")
```

Comment:
Hey Brandon - just confirming, are dates returned by the API **_always_** formatted as %d-%m-%y ?

Reply:
Yes, i think this line is mainly to convert the column to date (default is char) if datetime_type = true for plotting.

Reply:
Yes! We should include this in all functions to allow users to return the dates as datetimes. The date chars returned by the API are all %d-%m-%y